### PR TITLE
raidboss: convert 05-shb/raid to TS

### PR DESCRIPTION
**Pull request by quisquous**

### CODE REVIEW

File: ui/raidboss/data/05-shb/raid/e9s.ts [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
 import Outputs from '../../../../../resources/outputs';
+import { callOverlayHandler } from '../../../../../resources/overlay_plugin_api';
 import { Responses } from '../../../../../resources/responses';
 import ZoneId from '../../../../../resources/zone_id';
+import { RaidbossData } from '../../../../../types/data';
+import { PluginCombatantState } from '../../../../../types/event';
+import { TriggerSet } from '../../../../../types/trigger';
+
+// export type Data = RaidbossData;
+export interface Data extends RaidbossData {
```
Comment (panicstevenson) [inline ui/raidboss/data/05-shb/raid/e9s.ts:13]:
```suggestion
export interface Data extends RaidbossData {
```

File: ui/raidboss/data/05-shb/raid/e9s.ts [hunk 2]
```diff
@@ -173,18 +192,20 @@ const calculateSummonSafeZone = (boss, clone1, clone2, abilityId) => {
     .filter((pos) => !badZones.includes(pos))
     .map((pos) => directions[pos]);
 
-  if (safeZones.length !== 1)
+  const safeZone = safeZones[0];
+  if (safeZone === undefined || safeZones.length !== 1)
```
Comment (panicstevenson) [inline ui/raidboss/data/05-shb/raid/e9s.ts:196]:
```suggestion
  if (safeZones?.length !== 1)
```

**Review by panicstevenson — APPROVED** [review]

File: ui/raidboss/data/05-shb/raid/e4s.ts [hunk 3]
```diff
@@ -3,8 +3,17 @@ import NetRegexes from '../../../../../resources/netregexes';
...
+import { RaidbossData } from '../../../../../types/data';
+import { TriggerSet } from '../../../../../types/trigger';
 
-export default {
+// export type Data = RaidbossData;
+export interface Data extends RaidbossData {
+  playerMap?: { [name: string]: boolean };
+  phase?: string;
+  printedBury?: boolean;
+}
```
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/data/05-shb/raid/e4s.ts]: Thanks for catching all my sloppiness.  I don't know how I missed so many `// export type Data = RaidbossData`.

File: ui/raidboss/data/05-shb/raid/e12s.ts [hunk 4]
```diff
@@ -215,9 +233,13 @@ const intermediateRelativityOutputStrings = {
     ko: '에어로가',
   },
 };
+// TODO: eslint gets confused about `id` here in a type definition?
+// eslint-disable-next-line no-unused-vars
+type InterStrings = { [id in keyof typeof intermediateRelativityOutputStringsRaw]: LocaleText };
```
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/data/05-shb/raid/e12s.ts:238]: Fixed in #3304.

File: ui/raidboss/data/05-shb/raid/e9s.ts (hunk 2 above)
Comment (quisquous) [reply ui/raidboss/data/05-shb/raid/e9s.ts]: Good try on this one, but the `safeZone` is used below and so needs to be defined and have the right length.

**Pull request merged**